lib/screen/wallet_connect/wc_connect_page.dart

```diff
@@ -142,101 +142,116 @@ class _WCConnectPageState extends State<WCConnectPage>
     final wcConnectArgs = widget.wcConnectArgs;
     final beaconRequest = widget.beaconRequest;
 
+    late String typeConnect;
+    if (wc2Proposal != null) {
```

Comment:
I suggest a new design like this, do you think we can apply this:
- Create an abstract class for requests: eg: `ConnectionRequest` or `DappRequest`
- Inherit that class from 3 kinds of requests.
- Switch the type and cast it into the correct type to process the request.

it can reduce the redundant arguments passing into this screen and avoid null checks on the requests.